service/udp.go

```diff
@@ -201,19 +202,23 @@ func (s *udpService) Serve(clientConn net.PacketConn) error {
 				}
 				targetConn = nm.Add(clientAddr, clientConn, cipher, udpConn, clientLocation, keyID)
 			} else {
+				clientLocation = targetConn.clientLocation
+
 				unpackStart := time.Now()
 				textData, err := ss.Unpack(nil, cipherData, targetConn.cipher)
 				timeToCipher = time.Now().Sub(unpackStart)
 				if err != nil {
 					return onet.NewConnectionError("ERR_CIPHER", "Failed to unpack data from client", err)
 				}
 
+				// The key ID is known with confidence once decryption succeeds.
+				keyID = targetConn.keyID
```

Comment:
Where is this being used after the assignment?

Reply:
It's being used in the `defer` at line 160.